### ISSUE TITLE
[core] Generate json-schema is incorrect for http announcers

### ISSUE THREAD

**Issue by olivier-hubaut** [jreleaser/jreleaser#1211]
### Task List

- [x] Steps to reproduce provided
- [ ] Stacktrace (if present) provided
- [ ] Example that reproduces the problem (link to git repository is ideal)
- [x] Full description of the issue provided (see below)

### Steps to Reproduce

1. mvn jrealeaser:json-schema
2. Look at the definition of http announcer

### Expected Behaviour

Since the HTTP annourcer can have multiple targets, the json-schema should accept things like

```yaml
announce:
  http:
    xwiki: # first announcer
      # Here is the definition of that announcer
    confluence: # second announcer
      # Here is the definition of the second announcer
```

### Actual Behaviour

Since the json-schema defines the `http` property as `HttpAnnouncers` instead of `HttpAnnouncerMap`, it cannot be accept such structure. 

### Environment Information

- **Operating System**: Windows
- **JReleaser Version:** 1.4.0
- **JDK Version:** 17

**Comment by aalmiray** [comment]
Hmm this is strange. This is how it should be handled:

```yaml
announce:
  http:   # type is HttpAnnouncers
    xwiki: # type is HttpAnnouncer
      # Here is the definition of that announcer
    confluence: # type is HttpAnnouncer
      # Here is the definition of the second announcer
```

It seems to me that `httpAnnouncers` should allow all properties from `HttpAnnouncerMap` oiw these two types should merge.

**Comment by aalmiray** [comment]
Released in v1.5.0 -> https://github.com/jreleaser/jreleaser/releases/tag/v1.5.0
